node.go

```diff
@@ -306,6 +270,17 @@ func fieldRepetitionTypeOf(node Node) *format.FieldRepetitionType {
 	}
 }
 
+func fieldRepetitionTypeOf(node Node) format.FieldRepetitionType {
+	switch {
+	case node.Optional():
+		return format.Optional
+	case node.Repeated():
+		return format.Repeated
```

Comment:
Maybe more future proof:

```
case node.Required():
    return format.Required
default:
    panic("unknown node repetition type")
```

although maybe it's hard to think of what a fourth or fifth type would be.

Reply:
A risk is that the state may come from user input (say an exotic parquet file, or custom implementation of the Node interface), panic may not be the desired behavior then, but maybe validation should happen at a different stage as well... it seems hard to perceive what the right trade off would be.